src/plugins/intel_gna/frontend/scale_factor_calc.hpp

```diff
@@ -220,71 +220,81 @@ static double calculateWeightsReducerFromDstStats(Quantization dst_quant) {
+            (std::numeric_limits<uint16_t>::max() + 1ul);
+        auto maxSF = CalculateScaleFactorFromStats(levels, quantDataForInputLayer->_dst_quant.GetMinValues().front(),
+            quantDataForInputLayer->_dst_quant.GetMaxValues().front());
+        if (newOutputScale > maxSF) {
+            gnalog() << layer->name << ": Scale factor " << newOutputScale << " is too large. The maximum scale factor: "
+                << maxSF << " levels=" << levels << " min=" << quantDataForInputLayer->_dst_quant.GetMinValues().front()
+                << " max=" << quantDataForInputLayer->_dst_quant.GetMaxValues().front() << "\n";
+            return false;
         }
+    }
+    if (info.isActivation() || info.isConst()) {
+        gnawarn() << "[WARNING] requantize " << layer->name
+                    << ". Layer new output scale: " << newOutputScale
+                    << ", was " << quantDataForInputLayer->_dst_quant.GetScale() << std::endl;
+        quantDataForInputLayer->_dst_quant.SetScale(newOutputScale);
+        result = ScaleFactorUpdateResult(layer.get());
+        return true;
+    }
 
-        if (info.isWeightableIdentity() && !fp32eq(quantDataForInputLayer->_weights_quant.GetScale(), 1.0f)) {
-            auto reducer = std::max(1.0f, quantDataForInputLayer->_dst_quant.GetScale() / newOutputScale);
-            auto newWeightsScale = std::max(1.0f, quantDataForInputLayer->_weights_quant.GetScale() / reducer);
-            quantDataForInputLayer->_weights_quant.SetScale(static_cast<int32_t>(newWeightsScale));
-            quantDataForInputLayer->_dst_quant.SetScale(quantDataForInputLayer->_weights_quant.GetScale() *
-                quantDataForInputLayer->_src_quant.GetScale());
+    if (info.isWeightableIdentity() && !fp32eq(quantDataForInputLayer->_weights_quant.GetScale(), 1.0f)) {
+        auto reducer = std::max(1.0f, quantDataForInputLayer->_dst_quant.GetScale() / newOutputScale);
+        auto newWeightsScale = std::max(1.0f, quantDataForInputLayer->_weights_quant.GetScale() / reducer);
+        quantDataForInputLayer->_weights_quant.SetScale(static_cast<int32_t>(newWeightsScale));
+        quantDataForInputLayer->_dst_quant.SetScale(quantDataForInputLayer->_weights_quant.GetScale() *
+            quantDataForInputLayer->_src_quant.GetScale());
 
-            result = ScaleFactorUpdateResult(layer.get());
-            return true;
-        }
+        result = ScaleFactorUpdateResult(layer.get());
+        return true;
+    }
 
-        if (info.isFullyConnected() || info.isConvolution()) {
-            quantDataForInputLayer->_dst_quant.SetScale(newOutputScale);
-            quantDataForInputLayer->_weights_quant.SetScale(newOutputScale / quantDataForInputLayer->_src_quant.GetScale());
-            result = ScaleFactorUpdateResult(layer.get());
-            return true;
-        }
+    if (info.isFullyConnected() || info.isConvolution() || info.isPower()) {
```

Comment:
My general comment is that it's hard to follow changes in this PR without comments on what has changed and why (in code or in the ticket).
For example - how is this change connected with Add layer?

Reply:
> My general comment is that it's hard to follow changes in this PR without comments on what has changed and why (in code or in the ticket). For example - how is this change connected with Add layer?

I suppose changes are mostly connected to specific customer model, not just Add. For example, on your branch this change is present under commit "Accuracy fixes for PWL and Eltwise" https://github.com/sirzabek/openvino/commit/d6989b1a8bfac28a1aa1c47971f173fa1ea290c7